packages/node/src/policy-gen/policy-gen-compartment-map.js

```diff
@@ -10,73 +10,98 @@ import { DEFAULT_ENDO_OPTIONS } from '../compartment/options.js'
 import { defaultReadPowers } from '../compartment/power.js'
 import { ATTENUATORS_COMPARTMENT } from '../constants.js'
 import { log as defaultLog } from '../log.js'
-import { toEndoURL } from '../util.js'
+import { hrLabel, toEndoURL } from '../util.js'
 import { makePolicyGenCompartment } from './policy-gen-compartment-class.js'
 import { getCanonicalName } from './policy-gen-util.js'
 
 /**
- * @import {LoadCompartmentMapOptions} from '../internal.js'
- * @import {CompartmentMapDescriptor, Sources} from '@endo/compartment-mapper'
+ * @import {CompartmentDescriptorTransform, LoadCompartmentMapOptions, LoadCompartmentMapResult} from '../internal.js'
  */
 
 const { values } = Object
 
+/**
+ * This compartment descriptor transform replaces the `label` field of the
+ * compartment descriptor with the canonical name of the package.
+ *
+ * It is intended to be executed _last_ in the list of transforms.
+ *
+ * @type {CompartmentDescriptorTransform}
+ */
```

Comment:
What is the benefit of doing it this way instead of imperatively after other stuff, the way it was yesterday?

If there's benefits to this structure - great. If not, I'm worried it's more hidden now and harder to find out it's happening.

Reply:
useful, though? I mean, the end user can use a package manager and/or its lockfile to figure this out.

my reasoning for perhaps _not_ doing this would be that there will never be a `npm:foo>bar@1.0.0` and `npm:foo>bar@2.0.0` at once; one of these would need to be aliased to a different name (since dirs in `node_modules/*` refer to package names only).